File: Backend/BankingPaymentsApp_API/BankingPaymentsApp_API/Models/Query.cs
using System.ComponentModel.DataAnnotations;

namespace BankingPaymentsApp_API.Models
{
    public class Query
    {
        [Key]
        public int Id { get; set; }

        [Required(ErrorMessage = "Name is required!")]
        public string Name { get; set; }

        [Required(ErrorMessage = "Email is required!")]
        [EmailAddress(ErrorMessage = "Invalid email address.")]
        public string Email { get; set; }

        [Required(ErrorMessage = "Message is required!")]
        public string Message { get; set; }

        public string? Response { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: frontend/BankingPaymentsApp_Frontend/src/app/Components/landing-page/query/query.ts
import { Component } from '@angular/core';
import { FormBuilder, FormGroup, ReactiveFormsModule, Validators } from '@angular/forms';
import { QueryService } from '../../../Services/query.service';
import { Query } from '../../../Models/Query';
import { CommonModule } from '@angular/common';

@Component({
  selector: 'app-query',
  standalone: true,
   imports: [CommonModule, ReactiveFormsModule],
  templateUrl: './query.html',
  styleUrl: './query.css'
})
export class QueryComponent {

  queryForm: FormGroup;

  constructor(private fb: FormBuilder, private queryService: QueryService) {
    this.queryForm = this.fb.group({
      name: ['', Validators.required],
      email: ['', [Validators.required, Validators.email]],
      message: ['', Validators.required]
    });
  }

  submitQuery() {
    if (this.queryForm.invalid) {
      this.queryForm.markAllAsTouched();
      return;
    }

    const query: Query = this.queryForm.value;

    this.queryService.createQuery(query).subscribe({
      next: (res) => {
        alert('Your response has been submitted!');
        this.queryForm.reset();
      },
      error: (err) => {
        console.error(err);
        alert('Failed to submit your response. Please try again later.');
      }
    });
  }
}
